### PR TITLE
Remove unused code from ANCM setup and enable more MSVC warnings for SDL

### PR DESCRIPTION
**Pull request by adityamandaleeka**
This covers the installers part of https://github.com/dotnet/aspnetcore/issues/40525

### CODE REVIEW

**Review by HaoK — APPROVED** [review]

**Comment by adityamandaleeka** [comment]
Looks like this checks out. I'll push an update that removes the unused files.

**Comment by adityamandaleeka** [comment]
@wtgodbe I'd like to get your review on this too.

File: src/Installers/Windows/AspNetCoreModule-Setup/build/settings/common.props [hunk 1]
```diff
@@ -25,7 +25,9 @@
 
   <ItemDefinitionGroup>
     <ClCompile>
-      <WarningLevel>Level3</WarningLevel>
+      <!-- Microsoft.Security.SystemsADM.10086 - SDL required warnings -->
+      <WarningLevel>Level4</WarningLevel>
+      <TreatSpecificWarningsAsErrors>4018;4055;4146;4242;4244;4267;4302;4308;4509;4510;4532;4533;4610;4611;4700;4701;4703;4789;4995;4996</TreatSpecificWarningsAsErrors>
```
Comment (wtgodbe) [inline src/Installers/Windows/AspNetCoreModule-Setup/build/settings/common.props:30]: Add a comment about what these are? No need to enumerate all of them, maybe just a link to a doc/whatever lists the requirement

**Review by wtgodbe — APPROVED** [review]
LGTM other than the comment

File: src/Installers/Windows/AspNetCoreModule-Setup/build/settings/common.props (hunk 1 above)
Comment (adityamandaleeka) [reply src/Installers/Windows/AspNetCoreModule-Setup/build/settings/common.props:30]: The comment above covers that. Searching for that string publicly pulls up the relevant GitHub issue and internally takes you to the exact requirements.

Comment (adityamandaleeka) [reply src/Installers/Windows/AspNetCoreModule-Setup/build/settings/common.props:31]: Added link to issue too.

**Review by wtgodbe — APPROVED** [review]

**Pull request merged**